addons/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/HueBridge.java

```diff
@@ -998,24 +1005,25 @@ public void handleErrors(Result result) throws IOException, ApiException {
     private String enc(@Nullable String str) {
         try {
             if (str != null) {
-                return URLEncoder.encode(str, "utf-8");
+                return URLEncoder.encode(str, StandardCharsets.UTF_8.name());
             } else {
                 return "";
             }
         } catch (UnsupportedEncodingException e) {
-            // throw new EndOfTheWorldException()
             throw new UnsupportedOperationException("UTF-8 not supported");
         }
     }
 
     private String getRelativeURL(String path) {
-        String relativeUrl = "http://" + ip + "/api";
+        String basePath = "/api";
         if (username != null) {
-            relativeUrl += "/" + enc(username);
+            basePath += "/" + enc(username);
         }
-        if (!path.isEmpty()) {
-            relativeUrl += "/" + path;
+        try {
+            URI uri = new URI(protocol, null, ip, port, path.isEmpty() ? basePath : basePath + "/" + path, null, null);
+            return uri.toString();
+        } catch (URISyntaxException e) {
```

Comment:
Should we do some logging here? Its unlikely that the url is correct?

Reply:
Yes, I will. And I will do a second change: I will move the creation of the `URI` into the constructor. It is sufficient to resolve it once and not every time we want to connect to the Hue API.

Reply:
Done. But let me test it for at least one day before we merge.